### ISSUE TITLE
Missing support for atomic test and set operations

### ISSUE THREAD

**Issue by lmctv** [cannatag/ldap3#75]
From reading the available docs, I cannot find out how to replace the
following python-ldap script fragment:
```
	c = ldap.open('ldap.example.com')
	DN = 'cn=objectcn,dc=ldap,dc=example,dc=com'
	ATT = 'cn'
	OVAL = 'testValue'
	NVAL = 'setValue'
	c.modify_s(DN, [(ldap.MOD_ADD, ATT, OVAL),
			(ldap.MOD_DELETE, ATT, NVAL)])
```
which would translate to the following LDIF atomic change operation:
```
	dn: cn=objectcn,dc=ldap,dc=example,dc=com
	changetype: modify
	delete: cn
	cn: testValue
	-
	add: cn
	cn: setValue
```
While the change operations are sequenced, the modify change will happen atomically on the server.

For the use of the operation as a locking primitive, see:
 - http://www.openldap.org/lists/openldap-software/200009/msg00559.html (K. Zeilenga answers to a question about locking)
 - http://ldapwiki.willeke.com/wiki/LDIF%20Atomic%20Operations (LDIF Atomic Operations article from ldapwiki site)

**Comment by cannatag** [comment]
Please check the fix in dev,  it should work as requested. Now you can send a sequence of changes for a single attribute, refer to the docs and the test in testModifyOperation.py. Please note that atomicity must be provided by the ldap server.

Let me know if this fix works for you.

Bye,.Giovanni

**Comment by cannatag** [comment]
Released version 0.9.8.6 that includes this fix.

Bye,
Giovanni
